webknossos/webknossos/dataset/remote_dataset_registry.py

```diff
@@ -0,0 +1,87 @@
+from collections import defaultdict
+from typing import TYPE_CHECKING, DefaultDict, Dict, List, Mapping, TypeVar
+
+from boltons.dictutils import FrozenDict
+
+from webknossos.utils import LazyReadOnlyDict
+
+if TYPE_CHECKING:
+    from webknossos.dataset.dataset import RemoteDataset
+
+
+K = TypeVar("K")  # key
+V = TypeVar("V")  # value
+C = TypeVar("C")  # cache
+
+
+class RemoteOrganizationDatasetRegistry(LazyReadOnlyDict[str, "RemoteDataset"]):
+    """Dict-like class mapping dataset names to `RemoteDataset` instances."""
+
+    by_display_name: Mapping[str, "RemoteDataset"]
+    by_tag: Mapping[str, List["RemoteDataset"]]
+
+    def __init__(
+        self,
+        organization_id: str,
+        names: List[str],
+        by_display_name: Dict[str, str],
+        by_tag: Dict[str, List[str]],
+    ) -> None:
+        from webknossos.dataset.dataset import Dataset
+
+        super().__init__(
+            entries=dict(zip(names, names)),
+            func=lambda name: Dataset.open_remote(name, organization_id),
+        )
+        self.by_display_name = LazyReadOnlyDict(
+            entries=by_display_name,
+            func=lambda name: Dataset.open_remote(name, organization_id),
+        )
+        self.by_tag = LazyReadOnlyDict(
+            entries=by_tag,
+            func=lambda names: [
+                Dataset.open_remote(name, organization_id) for name in names
+            ],
+        )
+
+
+class RemoteDatasetRegistry(
```

Comment:
I'm not sure, we want to make access to all public datasets that easy. I think some users make datasets public, but think they are "unlisted". This would expose all of them. We don't need to close down the API, but only provide library functions for accessing a specified organization (the user's organization by default).

Reply:
Does the API already have an organization filter? @fm3

Reply:
I could also easily make the dict a function, so it's not possible to iterate over the keys anymore, e.g.
`list_remote_datasets(organiztation_id: Optional[str]=None)`, where the `None` default indicates to use the current organization.

Reply:
Ideally, we wouldn't even load the datasets from other orgs in this API call.

Reply:
> Ideally, we wouldn't even load the datasets from other orgs in this API call.

This is not possble atm, I added an issue for it: https://github.com/scalableminds/webknossos/issues/6329

I filtered by the current organization now by default in the client.